ThunderCloud/StormLoginViewController.swift

```diff
@@ -0,0 +1,269 @@
+			NotificationCenter.default.addObserver(forName: $0, object: nil, queue: .main, using: { [weak self] (notification) in
+				self?.updateBottomConstraint(with: notification)
+			})
+		})
+		
+		guard !OnePasswordExtension.shared().isAppExtensionAvailable() else {
+			return
+		}
+		
+		onePasswordButton.isHidden = true
+		passwordField.rightInset = 8
+	}
+	
+	override func viewWillAppear(_ animated: Bool) {
+		
+		super.viewWillAppear(animated)
+		
+		UIView.animate(withDuration: 1.0, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 0.0, options: [], animations: {
+			
+			self.backgroundView.alpha = 1.0
+			self.containerView.alpha = 1.0
+			
+		}, completion: nil)
+	}
+	
+	private func updateBottomConstraint(with keyboardNotification: Notification) {
+		
+		let hidden = keyboardNotification.name == .UIKeyboardWillHide
+		
+		guard let endFrameValue = keyboardNotification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? CGRect else { return }
+		guard let duration = keyboardNotification.userInfo?[UIKeyboardAnimationDurationUserInfoKey] as? TimeInterval else { return }
+		guard let curveValue = keyboardNotification.userInfo?[UIKeyboardAnimationCurveUserInfoKey] as? UInt else {  return }
+		let curveOptions = UIViewAnimationOptions(rawValue: curveValue << 16)
+		
+		self.bottomConstraint.constant = hidden ? 12 : endFrameValue.height + 12
+		
+		UIView.animate(withDuration: duration, delay: 0, options: curveOptions, animations: {
+			self.view.layoutIfNeeded()
+		}, completion: nil)
+	}
+	
+	/// Handles when the user has sucessfully logged in
+	private func handleLoggedIn() {
+		
+		// Just in-case!
+		loginError = nil
+		
+		// If we don't have a success view controller, simply dismiss calling completion
+		guard let sucessViewController = successViewController else {
+			dismissCallingCompletion()
+			return
+		}
+		
+		// If success VC doesn't have a view we should probably just dismiss,
+		// otherwise we'll get a broken UI
+		guard let childView = sucessViewController.view else {
+			dismissCallingCompletion()
+			return
+		}
+		
+		childView.alpha = 0.0
+		addChildViewController(sucessViewController)
+		containerView.addSubview(childView)
+		
+		if let gestureRecognizer = backgroundView.gestureRecognizers?.first {
+			backgroundView.removeGestureRecognizer(gestureRecognizer)
+		}
+		
+		// Remove translatesAutoresizingMaskIntoConstraints so we can constrain it to our container view
+		childView.translatesAutoresizingMaskIntoConstraints = false
+		
+		// Constrain the success view controller to our container view
+		containerView.attachEdges(to: childView)
+		
+		UIView.animate(withDuration: 1.0, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 0.0, options: [], animations: {
+			
+			self.titleLabel.alpha = 0.0
+			self.reasonLabel.alpha = 0.0
+			self.passwordField.alpha = 0.0
+			self.usernameField.alpha = 0.0
+			self.loginButton.alpha = 0.0
+			
+			childView.alpha = 1.0
+			
```

Comment:
Same for this one too, the one at the top is fine but it doesn't seem like other blocks have one at the end? What is preferred? :)

Reply:
I tend to put a line break if there is another opening brace on the same line as a closing one:
```
if condition {

    print("Do something")

} else {
    
    print("Do something else")
}
```